partner_event/wizard/res_partner_register_event.py

```diff
@@ -26,6 +28,25 @@ def _prepare_registration(self, partner):
     def button_register(self):
         registration_obj = self.env['event.registration']
         partner_obj = self.env['res.partner']
+        errors = []
         for partner_id in self.env.context.get('active_ids', []):
             partner = partner_obj.browse(partner_id)
-            registration_obj.create(self._prepare_registration(partner))
+            try:
+                with self.env.cr.savepoint():
+                    registration_obj.create(
+                        self._prepare_registration(partner))
+            except:
+                errors.append(partner.name)
```

Comment:
I think it would be nice if you included the error message too, to help user know the reason for it.

Also better if you include a comment on why do you expect that this could fail.